src/open_inwoner/react/lib/web-component/loader.tsx

```diff
@@ -0,0 +1,60 @@
+import {
+  createContextsForComponent,
+  findWebComponentsOnPage,
+  runAfterLoadHooks,
+  runBeforeLoadHooks,
+  runErrorHooks,
+  wcRegistry,
+  WebComponentKey,
+} from '.';
+
+/**
+ * Load a single web component with plugin lifecycle
+ */
+export const wcLoader = async (name: WebComponentKey): Promise<void> => {
```

Comment:
I think this can be private, making the API easier to follow. `registerWebComponents` is the public entrypoint.

```suggestion
const wcLoader = async (name: WebComponentKey): Promise<void> => {
```

Reply:
It might be better readable to create registerWebComponents as a class instead of function based.